src/testdrive/src/action/kafka/ingest.rs

```diff
@@ -61,6 +54,21 @@ pub fn build_ingest(mut cmd: BuiltinCommand) -> Result<IngestAction, String> {
         "bytes" => Format::Bytes,
         f => return Err(format!("unknown message format: {}", f)),
     };
+    let key_schema = cmd.args.opt_string("key-schema");
+    let key_serializer = cmd.args.opt_string("key-serializer");
+    if key_schema.is_some() && key_serializer.is_some() {
+        return Err("key-schema and key-serializer cannot both be defined".to_string());
+    }
+    let key_format = if let Some(schema) = key_schema {
+        Some(Format::Avro { schema })
+    } else if let Some(serializer_type) = key_serializer {
+        if serializer_type.to_lowercase() != "string" {
+            return Err(format!("invalid key-serializer: {}", serializer_type));
+        }
+        Some(Format::Text)
+    } else {
+        None
+    };
```

Comment:
Let's make this work the same way as `format` and `schema`. So `key-format=avro` requires that you also specify `key-schema`, and `key-format=bytes` doesn't require additional options.